lib/baremetalbasetest.pm

```diff
@@ -0,0 +1,269 @@
+=cut
+sub set_net_boot {
+    if (check_var('BACKEND', 'ipmi')) {
+        while (1) {
+            my $stdout = ipmitool('chassis bootparam get 5');
+            last if $stdout =~ m/Force PXE/;
+            ipmitool("chassis bootdev pxe");
+            sleep(3);
+        }
+    } else {
+        die('Backend ' . get_var('BACKEND') . 'not supported');
+    }
+}
+
+=head2 wait_boot
+
+wait_boot()
+
+Wait for the machine to boot up.
+=cut
+sub wait_boot {
+    my $timeout = shift;
+    if (!defined $timeout) {
+        $timeout = 1800;
+    }
+    if (check_var('BACKEND', 'ipmi')) {
+        select_consolei('sol', await_console => 0);
+        assert_screen('linux-login', $timeout);
+    } else {
+        die('Backend ' . get_var('BACKEND') . 'not supported');
+    }
+}
+
+=head2 host_lock
+
+ host_lock() 
+
+Lock the SUT from the baremetal support service. 
+=cut
+sub host_lock {
+    my $host        = get_required_var('SUT_IP');
+    my $ip          = inet_ntoa(inet_aton($host));
+    my $http_server = get_required_var('IPXE_HTTPSERVER');
+    my $url         = "$http_server/v1/host_lock/lock/$ip";
+
+    my $retries = get_var('BAREMETAL_LOCK_RETRIES', 60);
+    if (!check_var('BAREMETAL_LOCK_TOKEN'), '') {
+        diag('baremetalbasetest::host_lock(): Host is already locked, doing nothing');
+        return;
+    }
+    do {
+        my $response = HTTP::Tiny->new->request('GET', $url, {content => '', headers => {'content-type' => 'text/plain'}});
+        if ($response->{status} == 200) {
+            set_var('BAREMETAL_LOCK_TOKEN', $response->{content});
+            break;
+        } elsif ($response->{status} == 412) {
+            if ($retries != 0) {
+                $retries--;
+                sleep(60);
+            } else {
+                record_info('lock timeout', 'failed to acquire the host lock. giving up!', 'fail');
+                die('timeout acquiring the host lock - host is not being freed in time');
+            }
+        } else {
+            record_info('lock failure', "failed to acquire the host lock. got HTTP status $response->{status} and message $response->{reason}", 'fail');
+            die("baremetalbasetest::host_lock(): failed to acquire the host lock. got HTTP status $response->{status} and message $response->{reason}");
+        }
+    }
+
+}
+
+=head2 host_is_locked
+
+ host_is_locked() 
+
+Check if the SUT is locked in the support service
+=cut
+sub host_is_locked {
+    my $host        = get_required_var('SUT_IP');
+    my $ip          = inet_ntoa(inet_aton($host));
+    my $http_server = get_required_var('IPXE_HTTPSERVER');
+    my $url_status  = "$http_server/v1/host_lock/lock_state/$ip";
+
+    my $response = HTTP::Tiny->new->request('GET', $url_status, {content => '', headers => {'content-type' => 'text/plain'}});
+    if ($response->{status} != 200) {
+        diag("baremetalbasetest::host_is_locked(): unable to get lock state for $host: $response->{status} ($response->{reason}");
```

Comment:
we should `return 1 != 0;` other wise perl return last command return value which would be the on of `diag()` "maybe".